### ISSUE TITLE
test_python_reference_meta_functions takes too long to run

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#76733]
### 🐛 Describe the bug

Currently, these tests take nearly two minutes to run on CPU:

```
$ python test/test_ops.py -k test_python_reference_meta_functions                                                                                                              
...............................................................................................................................
...............................................................................................................................
...............................................................................................................................
......................................................................................................                         
----------------------------------------------------------------------                                                         
Ran 483 tests in 117.137s 
```

This is far too long for 483 tests; we're spending nearly 0.2s a test. It would be good to understand why these tests take so long.

### Versions

master

**Comment by mruberry** [comment]
Well, it is comparing all the Python references to their referenced operators, and it is comparing them on the reference inputs, which is an expanded set of tensors, and the references are implemented inefficiency w.r.t. eager mode... I don't know if these tests are taking that long considering all that?

**Comment by ezyang** [comment]
Part of the waste feels from cross product of configurations. Individually, it doesn't seem so bad to check all operators, or to check all dtypes, or to check expanded references, but multiply them all together and it ggets very slow.

**Comment by mruberry** [comment]
> Part of the waste feels from cross product of configurations. Individually, it doesn't seem so bad to check all operators, or to check all dtypes, or to check expanded references, but multiply them all together and it ggets very slow.

Yep. But the alternative is to test less and we would need to be very clever about how to test less, and then the system would be more confusing, too, as we might change, like, int16 behavior for some reason and think it's great but we decided not to test int16 anymore and so who knows what's happening.

**Comment by ezyang** [comment]
Yeah. If we treat the test as a black box, if you really want to be sure, you have no recourse but to test all the combinations. What I would like is if we took more of a white box approach, where we have some way of seeing into the implementation to see what implementation is being used (there's a fixed set), and then make sure we only do one representative operator per implementation.

**Comment by mruberry** [comment]
Yeah I would also like to move in that direction although sometimes the interactions between components are very tricky to account for, too.
